src/openeo_test_suite/tests/processes/processing/test_example.py

```diff
@@ -74,22 +89,30 @@ def test_process(
     # prepare the arguments from test JSON encoding to internal backend representations
     # or skip if not supported by the test runner
     try:
-        arguments = prepare_arguments(example["arguments"], id, connection, file)
+        arguments = _prepare_arguments(
+            arguments=example["arguments"],
+            process_id=process_id,
+            connection=connection,
+            file=file,
+        )
     except Exception as e:
-        # TODO: skipping on a generic `Exception` is very liberal and might hide real issues
-        pytest.skip(str(e))
+        # TODO: originally there was a `pytest.skip()` here, but that was too liberal, hiding real issues.
+        #       On what precise conditions should we skip? e.g. NotImplementedError?
+        # pytest.skip(str(e))
+        raise
 
-    throws = bool(example["throws"]) if "throws" in example else False
+    throws = bool(example.get("throws"))
     returns = "returns" in example
 
     # execute the process
     try:
-        result = connection.execute(id, arguments)
+        result = connection.execute(process_id, arguments)
     except Exception as e:
         result = e
 
     # check the process results / behavior
     if throws and returns:
+        # TODO what does it mean if test can both throw and return?
```

Comment:
See process test documentation in https://github.com/Open-EO/openeo-processes/pull/468

> If both `returns` and `throws` are provided, it means that either of it should be true.
> So either the process retuens as specified or throws the specified error.

This happens e.g. for division by zero where some implementations throw an exceptions, some return ±infinity.